puffin/src/trace.rs

```diff
@@ -201,15 +197,15 @@ impl<PB: ProtocolBehavior> KnowledgeStore<PB> {
     /// If we want a variable with lower specificity, then we can just query less specific
     pub fn find_variable(
         &self,
-        query_type_shape: TypeShape,
-        query: &Query<PB::Matcher>,
-    ) -> Option<&(dyn VariableData)> {
+        query_type_shape: TypeShape<PT>,
+        query: &Query<PT::Matcher>,
+    ) -> Option<&(dyn VariableData<PT>)> {
         let query_type_id: TypeId = query_type_shape.into();
 
-        let mut possibilities: Vec<Knowledge<PB::Matcher>> = self
+        let mut possibilities: Vec<Knowledge<PT>> = self
             .raw_knowledge
             .iter()
-            .filter(|raw| (query.source.is_none() || query.source.as_ref().unwrap() == &raw.source))
+            .filter(|raw| (query.source == None || query.source.as_ref().unwrap() == &raw.source))
```

Comment:
Keep `.is_none()`. It's [best practice](https://rust-lang.github.io/rust-clippy/master/index.html#/partialeq_to_none).